### PR TITLE
Dynamic Buffer using Vector

### PR DESCRIPTION
**Pull request by ItamarB-DI**

### CODE REVIEW

File: DynamicBuffer/Ex1.hpp [hunk 1]
```diff
@@ -64,60 +45,20 @@ DynamicBuffer<T>& DynamicBuffer<T>::operator=(DynamicBuffer<T>&& other) noexcept
...
-DynamicBuffer<T>::~DynamicBuffer() noexcept {
-
-    deAllocate();
-}
-
-
 template <typename T>
 T& DynamicBuffer<T>::operator[](int idx) {
 
-    if (idx < 0) {
```
Comment (idoa-DI) [inline DynamicBuffer/Ex1.hpp:87]: why did you remove the boundary check?
what would happen if idx is less than 0?
what would happen if idx is greater than `size`?

File: DynamicBuffer/Ex1.hpp [hunk 2]
```diff
@@ -64,60 +45,20 @@ DynamicBuffer<T>& DynamicBuffer<T>::operator=(DynamicBuffer<T>&& other) noexcept
 
     if (this != &other) {
```
Comment (idoa-DI) [inline DynamicBuffer/Ex1.hpp]: do we need to implement move assignment operator?

File: DynamicBuffer/Ex1.hpp [hunk 3]
```diff
@@ -11,50 +11,31 @@ class DynamicBuffer {
...
-    m_buffer = other.m_buffer;
-
-    other.m_buffer = nullptr;
-    other.m_size = 0;
-}
+DynamicBuffer<T>::DynamicBuffer(DynamicBuffer<T>&& other) noexcept : m_buffer(std::move(other.m_buffer)) {}
 
 
 template <typename T> //Copy Assignment
 DynamicBuffer<T>& DynamicBuffer<T>::operator=(const DynamicBuffer<T>& other) {
```
Comment (idoa-DI) [inline DynamicBuffer/Ex1.hpp]: do we need to implement copy assignment operator?

File: DynamicBuffer/Ex1.hpp (hunk 1 above)
Comment (ItamarB-DI) [reply DynamicBuffer/Ex1.hpp:87]: totally right should be using the  "at" method, so there wont be undefined behavior

**Review by idoa-DI — APPROVED** [review]
